### PR TITLE
pakcs: fixed for darwin

### PR DESCRIPTION
**Pull request by ghost**
###### Motivation for this change

Fix and build PAKCS and dependencies on macOS.
###### Things done
- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
  - [x] NixOS
  - [x] OS X
  - [ ] Linux
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/compilers/swi-prolog/default.nix [hunk 1]
```diff
@@ -15,20 +15,30 @@ stdenv.mkDerivation {
...
   configureFlags = "--with-world --enable-gmp --enable-shared";
 
   buildFlags = "world";
 
+  # still not fixed in upstream: abort trap 6 when called through
+  # symlink on macos, so wrap binary
+  postInstall = stdenv.lib.optionalString stdenv.isDarwin ''
+    local prog="$out/bin/swipl"
+    local hidden="$(dirname "$prog")/.$(basename "$prog")"-wrapped
+    mv $prog $hidden
```
Comment (copumpkin) [inline pkgs/development/compilers/swi-prolog/default.nix]: Doesn't `makeWrapper` already take care of hiding the original for you?

Comment (LnL7) [inline pkgs/development/compilers/swi-prolog/default.nix]: Yes, I think this should be `wrapProgram $out/bin/swipl`

Comment (gnidorah) [inline pkgs/development/compilers/swi-prolog/default.nix]: @copumpkin @LnL7 wrapProgram additionaly adds '-a "$0"' to exec and that breaks PAKCS build: https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/setup-hooks/make-wrapper.sh

**Comment by gnidorah** [comment]
Merge please? This passes `nix-shell -p nox --run "nox-review wip"` on both 10.11 and 10.12 (after fix-sierra).

File: pkgs/development/compilers/swi-prolog/default.nix (hunk 1 above)
Comment (copumpkin) [inline pkgs/development/compilers/swi-prolog/default.nix]: Oh but I see now that you already did that so I'll just merge, thanks!

Comment (gnidorah) [inline pkgs/development/compilers/swi-prolog/default.nix]: @copumpkin Thank you!

**Pull request merged**